### ISSUE TITLE
Add `Assert.Contains({expr1}.Where({expr2})` -> `Assert.Contains({expr1}, {expr2})` analyzer / fixer

### ISSUE THREAD

**Issue by dougbu** [xunit/xunit#1510]
Add an analyzer to warn about calls to single-argument `Assert.Contains(...)`, `Assert.DoesNotContain(...)` or `Assert.Single(...)` with an argument ending with a Linq `Where(...)` call.

Fixer should map the single-argument calls to the two-argument form. Example mapping:
``` c#
Assert.Contains(list.OfType<string>().Where(a => !string.IsNullOrEmpty(a)));
...
Assert.Contains(list.OfType<string>(), a => !string.IsNullOrEmpty(a));
```

Optionally could do the same thing with `Assert.Empty(...)` and `Assert.NotEmpty(...)`, mapping them to `Assert.DoesNotContain(...)` and `Assert.Contains(...)`, respectively. These methods are slightly more complicated to check because their `IEnumerable` arguments do not automatically implement `IEnumerable<T>`.

**Comment by dougbu** [comment]
Part of the #1504 split.

**Comment by bradwilson** [comment]
All three fixes are now available as part of `1.16.0-pre.24` https://xunit.net/docs/using-ci-builds
